src/QsCompiler/CompilerOptimizations/VariableRenaming.fs

```diff
@@ -0,0 +1,124 @@
+module Microsoft.Quantum.QsCompiler.CompilerOptimization.VariableRenaming
+
+open System.Collections.Immutable
+open System.Text.RegularExpressions
+open Microsoft.Quantum.QsCompiler.DataTypes
+open Microsoft.Quantum.QsCompiler.SyntaxExtensions
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Microsoft.Quantum.QsCompiler.Transformations.Core
+
+open ComputationExpressions
+open Types
+open Utils
+
+
+/// The ScopeTransformation used to ensure unique variable names
+type VariableRenamer(argTuple: QsArgumentTuple) =
```

Comment:
Are there any tests for this? It would be good to add an extensive set of tests.

Reply:
Added to todo list